### PR TITLE
feat: auth

### PR DESCRIPTION
**Pull request by tusharkandpal**
To check:

- `authSlice.js` for authentication i.e login & signup
- `displaySlice.js` for toggling auth tabs & sidebar in small screens
- `RequiresAuth.js` for private routes
- added `logInService.js` & `signUpService.js`

### CODE REVIEW

File: src/component/shared/nav/Nav.jsx [hunk 1]
```diff
@@ -1,10 +1,19 @@
-import { GiHamburgerMenu } from "react-icons/gi";
 import { Link } from "react-router-dom";
+import { useSelector, useDispatch } from "react-redux";
+import { GiHamburgerMenu } from "react-icons/gi";
+import { toggleSidebar } from "../../../features/displaySlice";
+import { logOut } from "../../../features/authSlice";
```
Comment (t007rushi) [inline src/component/shared/nav/Nav.jsx:5]: barrel export

File: src/component/shared/sidebar/Sidebar.jsx [hunk 2]
```diff
@@ -9,8 +11,15 @@ import { RiMovieFill } from "react-icons/ri";
...
   return (
-    <aside className="hidden flex flex-col items-center basis-full md:basis-44 shrink-0 text-center bg-neutral-800">
+    <aside
+      className={`${
+        sidebarToggle ? "flex" : "hidden"
+      } md:flex flex-col items-center basis-full md:basis-44 shrink-0 text-center bg-neutral-800`}
+    >
       <NavLink
         to="/"
         className={({ isActive }) =>
```
Comment (t007rushi) [inline src/component/shared/sidebar/Sidebar.jsx:25]: this is repeating one 
you can put this active state outside make function for this 
and pass in className

**Review by t007rushi — APPROVED** [review]
folder structure and naming convention looks good
go ahead

**Review by paarthbajaj — APPROVED** [review]
Clean code and good use of classes. Appreciate it. You can merge.

File: src/component/shared/nav/Nav.jsx (hunk 1 above)
Comment (tusharkandpal) [reply src/component/shared/nav/Nav.jsx:5]: yes, will implement in next PR for sure !

File: src/component/shared/sidebar/Sidebar.jsx (hunk 2 above)
Comment (tusharkandpal) [reply src/component/shared/sidebar/Sidebar.jsx:25]: this too

**Pull request merged**